src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -478,6 +478,11 @@ namespace Microsoft.FSharp.Core
         let inline isinstPrim<'T>(x:obj) = (# "isinst !0" type ('T) x : obj #)
         let inline castclassPrim<'T>(x:obj) = (# "castclass !0" type ('T) x : 'T #)
         let inline notnullPrim<'T when 'T : not struct>(x:'T) = (# "ldnull cgt.un" x : bool #)
+           
+        let inline internal notnullFast (x: ^T) = 
```

Comment:
Once it works, to make it public, 

1. Create an RFC

2. Expose it in LanguagePrimitives (rather than exposing a new module) and call it something like ``LanguagePrimitives.IsNonNull``

Thanks
don